src/Serilog/Capturing/PropertyBinder.cs

```diff
@@ -34,9 +34,13 @@ public PropertyBinder(PropertyValueConverter valueConverter)
     /// represented in the message template.</param>
     /// <returns>A list of properties; if the template is malformed then
     /// this will be empty.</returns>
-    public EventProperty[] ConstructProperties(MessageTemplate messageTemplate, object?[]? messageTemplateParameters)
+#if FEATURE_SPAN
+    public EventProperty[] ConstructProperties(MessageTemplate messageTemplate, ReadOnlySpan<object?> messageTemplateParameters)
+#else
+    public EventProperty[] ConstructProperties(MessageTemplate messageTemplate, object?[] messageTemplateParameters)
+#endif
     {
-        if (messageTemplateParameters == null || messageTemplateParameters.Length == 0)
+        if (messageTemplateParameters.Length == 0)
```

Comment:
May be worth including a few quick speculative tests to check that removing the `== null` test here is safe even when callers without strict null checking enabled call various logger methods, e.g.:

```csharp
[Fact]
public void NullMessageTemplateParametersDoNotBreakBinding()
{
    var log = new LoggerConfiguration().WriteTo.Sink(new CollectingSink()).CreateLogger();
    log.Information("test", (object?[]?) null);
    log.BindMessageTemplate("test", (object?[]?)null, out _, out _);
}
```

?